checker/src/block_visitor.rs

```diff
@@ -335,7 +335,20 @@ impl<'block, 'analysis, 'compilation, 'tcx, E>
         // Continue to deal with the branch targets.
         let discr = discr.as_int_if_known().unwrap_or(discr);
         for i in 0..values.len() {
-            let val: Rc<AbstractValue> = Rc::new(ConstantDomain::U128(values[i]).into());
+            let param_env = self.bv.type_visitor.get_param_env();
+            let switch_ty_and_layout = self
+                .bv
+                .tcx
+                .layout_of(param_env.and(switch_ty))
+                .expect("rustc should ensure this");
+            let val: Rc<AbstractValue> = if switch_ty.is_signed() {
+                Rc::new(
+                    ConstantDomain::I128(sign_extend(values[i], switch_ty_and_layout.size) as i128)
+                        .into(),
+                )
+            } else {
+                Rc::new(ConstantDomain::U128(values[i]).into())
+            };
```

Comment:
This code sequence is starting to look familiar. Perhaps factor it out and re-use it?